GameServerLib/GameObjects/Spell/Missile/SpellCircleMissile.cs

```diff
@@ -187,60 +187,81 @@ protected override bool IsValidTarget(IAttackableUnit unit)
                 return false;
             }
 
-            if (unit.Team == CastInfo.Owner.Team && !SpellOrigin.SpellData.Flags.HasFlag(SpellDataFlags.AffectFriends))
-            {
-                return false;
-            }
+            bool valid = true;
 
-            if (unit.Team == TeamId.TEAM_NEUTRAL && !SpellOrigin.SpellData.Flags.HasFlag(SpellDataFlags.AffectNeutral))
+            if (unit.Team == CastInfo.Owner.Team && !SpellOrigin.SpellData.Flags.HasFlag(SpellDataFlags.AffectFriends))
             {
-                return false;
+                valid = false;
```

Comment:
I think you can return `false` here instead of continuing checking (not sure that C# can see this optimization very easily).